content/en/blog/2022/instrument-kafka-clients/index.md

```diff
@@ -0,0 +1,252 @@
+---
+title: Instrumenting Apache Kafka clients with OpenTelemetry
+linkTitle: Instrumenting Apache Kafka clients with OpenTelemetry
+date: 2022-09-05
+author: "[Paolo Patierno](https://github.com/ppatierno)"
+---
+
+Nowadays, [Apache Kafka](https://kafka.apache.org/) is chosen as the central piece in a distributed system.
+Different services communicate each other by using Apache Kafka as a messaging system but even more as en event or data streaming platform.
+
+Taking into account the cloud-native approach for developing microservices, quite often [Kubernetes](https://kubernetes.io/) is also used to run the workloads.
+In this scenario, you can also easily deploy and manage an Apache Kafka cluster on top of it, by using a project like [Strimzi](https://strimzi.io/).
+It takes care of the overall Kafka infrastructure, while you can focus on developing applications which use it.
+
+Within the overall picture, because of the distributed nature, it is quite difficult to track how messages are moved around.
+This is where the OpenTelemetry project comes into the picture.
+It defines the specification of semantic conventions for [messaging systems](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md) as well as it provides multiple instrumentation libraries for adding tracing to messaging based applications.
```

Comment:
Although this it the otel blog, the reader might not know what semantic conventions are. Maybe skip that for now and rephrase that to:

This is where OpenTelemetry comes into the picture: it provides multiple instrumentation libraries for adding tracing to messaging based applications. Of course, there is one for Apache Kafka clients.

Reply:
I thought it was useful to mention the messaging semantic conventions because I am referring to them in the `Instrumentation in action` section as well [here](https://github.com/open-telemetry/opentelemetry.io/pull/1677/files#diff-1713a18cb9f06d7587ef469a52bb57100157ee894ee5a3aa6bf40ecd46147797R277), to explain that current "output" in terms of trace, spans and relationship between them is evolving and going to change with the OTEP.

Reply:
then put the instrumentation first and the semantic conventions second? People are more interested in the first I would say.